### PR TITLE
#3: Implement and bind data model classes

### PR DESCRIPTION
**Pull request by pierrepebay**
Fixes: #3

### CODE REVIEW

**Comment by pierrepebay** [comment]
Implemented `Block` data model class, needs type checking in constructor (using nanobind [`nb::type_error`](https://nanobind.readthedocs.io/en/latest/api_core.html#_CPPv4N8nanobind10type_errorE), as is used in `detach_object_id`).

**Comment by ppebay** [comment]
@pierrepebay a reference example (from the `step.yaml` case) regarding our discussion of 05/03 pertaining to the problem of linear scaling of surface area w.r.t. widely varying object loads (as opposed to, for instance, logarithmic scaling which could be added in the future):

https://user-images.githubusercontent.com/17439260/236169138-64091525-6a01-4233-9afb-2c74cacf4049.mp4

Objects are present, and glyphed indeed in all phases, but hardly visible for phases post-initial one.

**Comment by pierrepebay** [comment]
When using the exported meshes in vtp in paraview, we can obtain a very simple first visualization:

| Current result when viewing in paraview | current LBAF exported visualization |
| ---------------------------------------- | ------------------------------------ |
|  | |

**Comment by pierrepebay** [comment]
@lifflander this PR is now ready for review, object communications will be in an other issue/PR.

File: examples/example2.cc [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+int main() {
+  using namespace vt;
+  using namespace tv;
+  // Read JSON file and input data
+
+  std::filesystem::path p = std::filesystem::path(SRC_DIR) / "tests/unit/lb_test_data" ;
+  std::string path = std::filesystem::absolute(p).string();
+
+  NodeType rank = 0;
```
Comment (lifflander) [inline examples/example2.cc:84]: Let's clean this up a bit with a for loop over the ranks.

File: lib_example.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import my_ext
```
Comment (lifflander) [inline lib_example.py:1]: Should we move this file to `vt-tv/examples` or maybe create a scripts directory?

File: src/vt-tv/render/render.h [hunk 3]
```diff
@@ -67,9 +67,61 @@
...
+#include <cstdlib>
+#include <tuple>
+#include <limits>
+#include <map>
+#include <unordered_set>
+#include <set>
 
 namespace vt { namespace tv {
 
+template <typename T>
```
Comment (lifflander) [inline src/vt-tv/render/render.h:95]: This would probably more efficiently and cleanly implemented with std::array and just forward the `[]` operator.

Comment (lifflander) [reply src/vt-tv/render/render.h:95]: In fact, given how it's used, you might just want to delete this and replace with std::array.

**Review by lifflander — CHANGES_REQUESTED** [review]

File: lib_example.py (hunk 2 above)
Comment (pierrepebay) [reply lib_example.py:1]: I'll move it to `vt-tv/examples/scripts`

File: src/vt-tv/render/render.h (hunk 3 above)
Comment (pierrepebay) [reply src/vt-tv/render/render.h]: I replaced it entirely with `std::array<typename, 3>`, and replaced the vectors I was using to store 3 elements with it as well.

**Comment by pierrepebay** [comment]
@lifflander I have addressed the requested changes, should be ready to merge!

**Review by lifflander — APPROVED** [review]

**Pull request merged**